### PR TITLE
added functionality to ordering posts, as well as split up props to use them better/more specifically

### PR DESCRIPTION
**Pull request by jhcoll**
…se them better/more specifically

### CODE REVIEW

File: src/components/posts/OrderPosts.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          </div>
+        </>
+      )}
+      <div className={classes.actions}>
+        {orderUpDown ? (
+          <button onClick={downHandler} style={{ backgroundColor: "blue" }}>
+            DOWN
+          </button>
+        ) : (
+          <button onClick={downHandler}>DOWN</button>
```
Comment (jluxford-scottlogic) [inline src/components/posts/OrderPosts.js:62]: Could maybe try and combine the `UP` and `DOWN` buttons into one button

File: src/components/posts/PostList.js [hunk 2]
```diff
@@ -1,22 +1,59 @@
...
+  function orderByDate() {
+    posts.sort(function (a, b) {
+      return new Date(b.date) - new Date(a.date);
+    });
+  }
+
+  function orderByName() {
+    posts.sort(function (a, b) {
+      return a.author.localeCompare(b.author);
+    });
```
Comment (jluxford-scottlogic) [inline src/components/posts/PostList.js:32]: Needing to have all of the sorting information in this class feels like quite a large gap between where it is changed and setup compared with where it is defined. Can you think of a way to not need these conditional cases and move a lot of this logic into the `OrderPosts` class

File: src/components/posts/buttons/DeleteButton.js [hunk 3]
```diff
@@ -1,16 +1,46 @@
...
         method: "DELETE",
       }
     ).then(() => {
-      props.setRefreshPost(true);
+      setShowDeleteAlert(false);
+      setPostDelete(false);
     });
   }
+  function deleteAlert() {
+    if (showDeleteAlert) {
```
Comment (jluxford-scottlogic) [inline src/components/posts/buttons/DeleteButton.js:20]: This `if` statement can be removed and reduced to a more concise function

**Review by jluxford-scottlogic — APPROVED** [review]
Nice set of changes, a few bits and places has some quite verbose code it might be worth going through and attempting a refactor at some point, where you can attempt to make it look a lot cleaner and keep the code quality high

File: src/components/posts/OrderPosts.js (hunk 1 above)
Comment (jhcoll) [reply src/components/posts/OrderPosts.js:62]: was thinking the same

File: src/components/posts/buttons/DeleteButton.js (hunk 3 above)
Comment (jhcoll) [reply src/components/posts/buttons/DeleteButton.js:20]: showDeleteAlert = !showDeleteAlert;

Comment (jhcoll) [reply src/components/posts/buttons/DeleteButton.js:20]: right?

**Pull request merged**